client_int_test.go

```diff
@@ -0,0 +1,289 @@
+// +build integration
+
+package function_test
+
+import (
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	boson "github.com/boson-project/func"
+	"github.com/boson-project/func/buildpacks"
+	"github.com/boson-project/func/docker"
+	"github.com/boson-project/func/knative"
+)
+
+/*
+ NOTE:  Running integration tests locally requires a configured test cluster.
+        Test failures may require manual removal of dangling resources.
+
+ ## Integration Cluster
+
+ These integration tests require a properly configured cluster,
+ such as that which is setup and configured in CI (see .github/workflows).
+ A local KinD cluster can be started via:
+   ./hack/allocate.sh && ./hack/configure.sh
+
+ ## Integration Testing
+
+ These tests can be run via the make target:
+   make test-integration
+  or manually by specifying the tag
+   go test -v -tags integration ./...
+
+ ## Teardown and Cleanup
+
+ Tests should clean up after themselves.  In the event of failures, one may
+ need to manually remove files:
+   rm -rf ./testdata/example.com
+ The test cluster is not automatically removed, as it can be reused.  To remove:
+   ./hack/delete.sh
+*/
+
+const (
+	// DefaultRegistry must contain both the registry host and
+	// registry namespace at this time.  This will likely be
+	// split and defaulted to the forthcoming in-cluster registry.
+	DefaultRegistry = "localhost:5000/func"
+
+	// DefaultNamespace for the underlying deployments.  Must be the same
+	// as is set up and configured (see hack/configure.sh)
+	DefaultNamespace = "func"
+)
+
+func TestList(t *testing.T) {
+	verbose := true
+
+	// Assemble
+	lister, err := knative.NewLister(DefaultNamespace)
```

Comment:
This is a great start, one small recommendation I would make is to reduce assumptions about the state of the cluster so that you can run tests over and over. An easy way to do this here would be to generate the names of namespaces so that you got a fresh one on each run.